File: frontend/src/types/productCategory.ts
export interface ProductCategory {
  id: string;
  name: string;
  group: string;
  description: string | null;
  isActive: boolean;
}

export interface ProductCategoryCreateRequest {
  name: string;
  group: string;
  description?: string;
}

export interface ProductCategoryUpdateRequest {
  name: string;
  group: string;
  description?: string;
  isActive: boolean;
}

export interface ProductCategoryQueryParams {
  name?: string;
  categoryGroup?: string;
  isActive?: boolean;
}

File: backend/src/main/java/vn/nguongocso/farm/entity/ProductCategory.java
package vn.nguongocso.farm.entity;

import java.util.UUID;

import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * Entity đại diện cho danh mục loại cây trồng.
 */
@Entity
@Table(name = "product_categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductCategory {
	@Id
	@Column(name = "id", nullable = false, updatable = false)
	@JdbcTypeCode(SqlTypes.CHAR)
	private UUID id;

	@Column(name = "name", nullable = false)
	private String name;

	@Column(name = "category_group")
	private String group;

	@Column(name = "description")
	private String description;

	@Column(name = "is_active", nullable = false)
	private Boolean isActive;
}